doc/snmp/snmp_ciscobgp4mib.md

```diff
@@ -0,0 +1,44 @@
+iso.3.6.1.4.1.9.9.187.1.2.5.1.3.1.4.10.0.0.1 = INTEGER: 6
+iso.3.6.1.4.1.9.9.187.1.2.5.1.3.2.16.252.0.0.0.0.0.0.0.0.0.0.0.0.0.0.1 = INTEGER: 6
+```
+
+## Design considerations for multi-asic platform
+### Extending current design
+In multi-asic platform, SNMP docker will be running on the host. BGP docker will be running per asic, in separate network namespace. If the currrent design has to be extended, then SNMP docker will have to connect to each BGP via TCP or unix socket. As each BGPd is in a different namespace, BGP docker in each namespace can interact with the snmp_ax_impl in host using docker0 bridge. Docker0 bridge has veth pairs with one interface of the pair on the host and another interface(eth0) inside the namespace. BGPd inside the BGP docker in namespace can open TCP socket to listen on eth0 IP address of the namespace instead of localhost. Snmp_ax_impl can then connect to TCP sockets on each namespace and retrieve data. Another option is to use UNIX socket /var/run/bgpd.vty. For TCP socket, Bgpd in each BGP docker will open socket 240.127.1.x 2065 to talk to VTYSH, currently socket is opened on localhost. If Unix socket is to be used, then /var/run/bgpd.vty of each BGP docker can be used by snmp_ax_impl to get the required data. The issue with this approach is that there will be N number of sockets opened for N asic platform. Also, each BGP docker will have to be updated to use the docker0 IP address network to open the socket.
+
+### New design proposal using STATE_DB
+To avoid multiple-socket approach, the data required by CiscoBgp4MIB can be populated by a new daemon in STATE_DB. This data can be retrieved by snmp_ax_impl from each namespace for multi-asic platform. Current implementaion of CiscoBgp4MIB in SONiC retrive the below information from the device:
+1. Neighbor IP address 
+2. Neighbor BGP state
+Currently snmp_ax_impl parses the data from Bgpd vtysh to get 'neighbor ip' and 'state'.
+
+Proposal is to add a new daemon called 'bgpmond' which will populate STATE_DB with the above infomration. 
+Schema:
+```
+NEIGH_STATE_TABLE {
+    "<neigh_ip>" { 
+        "State" : "active/connect/idle/etc"
+    }
+}
+```
+Currently, NEIGH_STATE_TABLE will be used by SNMP. This table can be used by telemetry or any other docker in future.
+
+### Bgpmond daemon to update STATE_DB
+This is the new daemon that runs inside of the BGP docker.  It will periodically (every 15 seconds) pull the bgp neighbor information by calling "show bgp summary json" and use the output to update the State DB accordingly.  In order to prevent unnecessary update to the State DB, a copy of each neighbor state is cached and used to detect if there are any changes from each newly pulled neighbor state.  Only when there is a change, then that particular entry is updated.  In a steady state situation, there is rarely a need to update the state DB.  If the neighbor is deleted from configuration, the corresponding state DB entry will also be cleaned up.
```

Comment:
>the [](start = 43, length = 3)

each

Reply:
@qiluo-msft 
Will fix the wording as following: "This is the new daemon that runs inside of each BGP docker..."

Reply:
Fixed as per comment.